src/constructs/core/identity.ts

```diff
@@ -1,5 +1,24 @@
-export interface Identity {
+import type { IConstruct } from "@aws-cdk/core";
+import { Tags } from "@aws-cdk/core";
+
+export interface StackStageIdentity {
   stack: string;
   stage: string;
+}
+
+export interface AppIdentity {
   app: string;
 }
+
+export interface Identity extends StackStageIdentity, AppIdentity {}
+
+export const AppIdentity = {
+  suffixText(appIdentity: AppIdentity, text: string): string {
+    const titleCaseApp = appIdentity.app.charAt(0).toUpperCase() + appIdentity.app.slice(1);
+    return `${text}${titleCaseApp}`;
+  },
+  addTag<T extends IConstruct>(appIdentity: AppIdentity, construct: T): T {
+    Tags.of(construct).add("App", appIdentity.app);
+    return construct;
```

Comment:
I find this return type a little surprising for a function called `addTag`

Reply:
Agreed. It's mainly out of laziness as we can do this:

```typescript
const snsTopic = props.existingSnsTopic?.externalTopicName
  ? Topic.fromTopicArn(
      scope,
      topicId,
      `arn:aws:sns:${scope.region}:${scope.account}:${props.existingSnsTopic.externalTopicName}`
    )
  : AppIdentity.addTag(
      props,
      new GuSnsTopic(scope, topicId, {
        overrideId: !!props.existingSnsTopic?.logicalIdFromCloudFormation,
      })
    );
```

The alternative is:

```typescript
const newTopic = () => {
  const topic = new GuSnsTopic(scope, topicId, {
    overrideId: !!props.existingSnsTopic?.logicalIdFromCloudFormation,
  });

  AppIdentity.addTag(props, topic);
  return topic;
};

const snsTopic = props.existingSnsTopic?.externalTopicName
  ? Topic.fromTopicArn(
      scope,
      topicId,
      `arn:aws:sns:${scope.region}:${scope.account}:${props.existingSnsTopic.externalTopicName}`
    )
  : newTopic();
```

Maybe `withTag` is a better name?

Reply:
Ah yes, I see the benefit now. We could go for `withTag` or `taggedResource`, but it's not a big deal so feel free to ignore this!

Reply:
Renamed to `taggedConstruct` to match the name of the parameter.